### ISSUE TITLE
Unexpected touch phase during forwarding

### ISSUE THREAD

**Issue by timtraversy** [flutter/flutter#27580]
This is admittedly not a high priority bug, but a Platform View can't be placed inside a Draggable without causing a bug.
Minimal repro:
```dart
  Widget build(BuildContext context) {
    final view = Container(
      height: 200,
      width: 200,
      child: WebView(
        initialUrl: 'https://google.com',
      ),
    );
    return Center(
      child: Draggable(
        child: view,
        feedback: view,
      ),
    );
  }
```
When run in Xcode, this throws the error `[Assert] Unexpected touch phase during forwarding`, and causes the Platform View to freeze and stop receiving touch events.

While my use case of dragging a Platform View may be uncommon, this could point to a larger bug in Platform View gesture recognition, so I'm dropping it here.

**Comment by kangwang1988** [comment]
@timtraversy 
Thanks for you feedback.
But I do think it would be better if you can give a minimal demo project which can reproduce this issue.

**Comment by timtraversy** [comment]
Sorry I was blocked on https://github.com/flutter/flutter/issues/27670 for most of today. Here's my demo project: https://github.com/timtraversy/draggable_platform.

**Comment by kangwang1988** [comment]
cc @xster 
In this case(drag failed in my mac&iphone), 

![Uploading Screen Shot 2019-02-09 at 11.51.17 PM.png…]()

**Comment by timtraversy** [comment]
Ok I'll take a look.

**Comment by xster** [comment]
@amirh for platform views

**Comment by javawag** [comment]
FYI this also happens when placing platform views inside of `PageView` widgets. Probably for the same reason as `Draggable`s but may make the issue higher priority as I imagine the use-case of platform views within `PageView`s might be more common?

**Comment by amirh** [comment]
In the example app linked above the webview has no key, when dragging starts the position of the webview in the widget tree is changed which makes it lose its state(and dispose the underlying platform view) which is causing the problem.

Verified that the following snippet(same code, just adding a key to the webview) works:
```dart
class _MyHomePageState extends State<MyHomePage> {
  final GlobalKey key = GlobalKey();
  @override
  Widget build(BuildContext context) {
    final view = Container(
      height: 200,
      width: 200,
      child: WebView(
        initialUrl: 'https://google.com',
        key: key,
      ),
    );
    return Center(
      child: Draggable(
        childWhenDragging: Container(),
        child: view,
        feedback: view,
      ),
    );
  }
}
```

@timtraversy I believe this should resolve the issue you were hitting, feel free to let me know if not or close this issue.

**Comment by timtraversy** [comment]
Thanks! I've moved on from this piece of functionality in my app, but if adding a key fixed the sample I sent you then I think this can be closed.

**Comment by javawag** [comment]
Don't know if this needs reopening, but adding `GlobalKey`s I can still reproduce this with `PageView` widget.

Reproducible snippet:
```
class _MyHomePageState extends State<MyHomePage> {
  final _webviewKeys = List<GlobalKey>.generate(10, (_) => GlobalKey());

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text("Test")),
      body: Center(    
        child: PageView.builder(
          itemCount: 10,
          itemBuilder: (ctx, i) {
            return WebView(
              initialUrl: "https://www.google.com",
              key: _webviewKeys[i],
            );
          }
        ),
      )
    );
  }
}
```

It seems to work okay on Android but on iOS I sometimes get the mentioned assert message but failing that I always get strange scrolling behaviour (page snapping breaks) when scrolling the webviews. If I add a `Padding` widget around the `WebView` and scroll from there instead it works as expected.

**Comment by wangfengnan** [comment]
same question while using PageView and WebView

**Comment by javawag** [comment]
Have opened a new issue for this, https://github.com/flutter/flutter/issues/29427.
